### DISCUSSION TITLE
How to put "use client" on mdx file?

### DISCUSSION THREAD

**Discussion by ervinpiol** [fuma-nama/fumadocs#405] [Q&A]
I am creating a design system I want to put the components but I don't know how to fix the "use client"

**Comment by fuma-nama** [comment]
Why? You can add the use client directive to your components and use them in MDX files.

**Reply by SySagar** [reply]
@fuma-nama  but still wewould require to putuse client in the parent component which are server by default. In my case too, although I added 'use client' in my components still it throws error that parents are not client component.

**Reply by fuma-nama** [reply]
Hmm can't get your point, could you elaborate a bit more?

**Reply by SySagar** [reply]
ok so assume I have a custom button , a client component. I am using monorepo so I imported it from my ui libray into docs files. I directly put it in the mdx with proper import. The page.tsx of docs is by default a serve componet right. But componenet is clinet, hence it throws error that parent is server and cannot put client.

Although I beleive it should work but still it shows this error. Note that I am using vite to bundle it up and then use it which is not taking use client into consideration ig.

**Reply by fuma-nama** [reply]
Why would you use Vite? If you are using Content Collections, you should pass components to the MDX component in page.tsx. Fumadocs MDX supports this natively

**Reply by SySagar** [reply]
no what I am saying is that I have a turborepo with directories: apps/ui and apps/docs(fuma-docs). Bacailly I am using a bundled version of my ui libray in the docs. However in bundle there is no use client directive which throws an error in the docs as it is a server compoenet. and no use client directive have been found. 

Now I imported it into index.ts suppsoe there I am putting 'use cleint' directive on top . However if you see the video below you can see how the ui is getting distorted. So I wanted to ask any fix to that?

**Comment by SySagar** [comment]
Ok So few realisations. When using bundlers the use client isn't getting used in my bundle file due to which it was still showing me that server componenet.

https://www.loom.com/share/61a961d31a944afd95210791cd549ebd?sid=6d1f91c6-9f01-4cdb-8027-3a19fd562a65

here I have used a technie wher I am exporting it first to a file and then using use client directive above it this way it works but for osme reason the ui is distorted. Can you help please?

**Reply by SySagar** [reply]
Please have a lo here @fuma-nama

**Reply by fuma-nama** [reply]
Cannot help without a reproduction

**Reply by SySagar** [reply]
umm how would you then suggest me to showcase my client components. I do have decalred the directives of use client but as I mentioed above it is not working :-(

p.s: If possible can you look into this issue. I am blocked actually here and can't move forward until thiangs work fine.

**Reply by fuma-nama** [reply]
Adding "use client" to your client components should works. Otherwise it's about other issues, I need more information on this

**Reply by SySagar** [reply]
But the use client thing is not working in my guess as I showed you in the video. Here you can see that I did have used the directive still it isn't working. Like the componenet does appears but the ui is distorted as you can see. If I remove my client component eveything works fine.

also note that the thing @groovy-box/ui is the bundled package . As per my understanding the bundler does not add use client that is why I made an index.ts file where I am basically exporting the components with 'use client' declared explicitly.

https://www.loom.com/share/c16094b6f2bf4a5dbfef42666e75f82e?sid=0b4d6c11-98ec-4c2b-aee4-742bcf40f7ec

**Reply by fuma-nama** [reply]
probably because of some styles of your components conflict with Fumadocs UI. Use our official plugin if you're using Tailwind CSS.

**Reply by SySagar** [reply]
I guess the plugin is used in tailwind config by default. What else I need to configure ?

Shoul I put it in my ui?

**Reply by fuma-nama** [reply]
If yes, use devtools to check the styles

**Reply by SySagar** [reply]
yes it does use tailiwnd. Is there a easy way so that fumadocs uses it original tailwind ocnfiguration rather than the one from my build. Althought its a bit strange to me how is it using it in the first place?

**Reply by SySagar** [reply]
ok so I added a prefix to my ui kit which solves most of the problmes but my own kit styles are gone none :(

**Reply by fuma-nama** [reply]
Fumadocs uses its own fd- prefix. Have you checked the conflicted styles?

**Reply by SySagar** [reply]
yes when I added a prefix of my own everything works fine. except that my own ui kit lost it styles.

I used this : https://www.npmjs.com/package/postcss-prefix-selector#usage-with-vite


HOw did u use it in foma-docs. If you can elaborate.

**Reply by SySagar** [reply]
For some reason, I am unable to handle this style clash. Is there any other way?

**Reply by fuma-nama** [reply]
You need to provide a reproduction

**Reply by SySagar** [reply]
hey @fuma-nama , i was able to solve this fianlly. however the only problem is that I have to manually add a lot of those tailwind prefix which is too much. Is there any easy way to auto the prefix?

**Reply by fuma-nama** [reply]
Just do it as you want.

I don't have any information about your component library and your Tailwind CSS design system, like how it is built? What configurations have you made? Which parts are conflicting with Fumadocs UI?

It's up to you to customise the styles, I don't have opinions on it.

**Reply by SySagar** [reply]
hi, just wanted to say everything works fine now. Just had to scope my designs and no conflcits. Again thanks for help ! :)
